pgjdbc/src/main/java/org/postgresql/core/v3/ConnectionFactoryImpl.java

```diff
@@ -775,20 +775,20 @@ private void doAuthentication(PGStream pgStream, String host, String user, Prope
               case AUTH_REQ_SASL:
                 LOGGER.log(Level.FINEST, " <=BE AuthenticationSASL");
 
-                String password = AuthenticationPluginManager.getPassword(AuthenticationRequestType.SASL, info);
+                char[] password = AuthenticationPluginManager.getPassword(AuthenticationRequestType.SASL, info);
                 if (password == null) {
                   throw new PSQLException(
                       GT.tr(
                           "The server requested SCRAM-based authentication, but no password was provided."),
                       PSQLState.CONNECTION_REJECTED);
                 }
-                if (password.equals("")) {
+                if (password.length == 0) {
                   throw new PSQLException(
                       GT.tr(
                           "The server requested SCRAM-based authentication, but the password is an empty string."),
                       PSQLState.CONNECTION_REJECTED);
                 }
-                scramAuthenticator = new org.postgresql.jre7.sasl.ScramAuthenticator(user, castNonNull(password), pgStream);
+                scramAuthenticator = new org.postgresql.jre7.sasl.ScramAuthenticator(user, castNonNull(String.valueOf(password)), pgStream);
```

Comment:
We should probably consider changing `ScramAuthenticator` to `char[]` as well

Reply:
Yes I mention that in the PR. It's self-contained so somebody can follow up with that separately.

Reply:
seems we should also refactor it to move it out of ...jre7.. as well
Is there any reason not to change it to char[] now ?

Reply:
I would agree with @sehrope , the key point for now is to get the API right == move to `char[]` in the interface.
The implementation can be adjusted later.

Reply:
So if you have no other issues should we merge this then ?

Reply:
Ultimately we need to call ongres/scram, and they don't seem to support `char[]` yet: https://gitlab.com/ongresinc/scram/-/issues/14

Reply:
@jorsol can you provide some insight/suppport here ?